test/meta_transaction_test.ts

```diff
@@ -0,0 +1,734 @@
+        salt: string;
+        makerAssetData: string;
+        takerAssetData: string;
+        makerFeeAssetData: string;
+        takerFeeAssetData: string;
+        signature: string;
+    }
+
+    function stringifyOrderBigNumbers(order: SignedOrder): StringifiedOrder {
+        return {
+            ...order,
+            makerAssetAmount: order.makerAssetAmount.toString(),
+            makerFee: order.makerFee.toString(),
+            takerAssetAmount: order.takerAssetAmount.toString(),
+            takerFee: order.takerFee.toString(),
+            salt: order.salt.toString(),
+            expirationTimeSeconds: order.expirationTimeSeconds.toString(),
+        };
+    }
+
+    interface QuoteTestCase {
+        quote: GetMetaTransactionQuoteResponse;
+        expectedBuyAmount: string;
+        expectedOrders: SignedOrder[];
+        expectedPrice: string;
+    }
+
+    function assertCorrectQuote(testCase: QuoteTestCase): void {
+        expect(testCase.quote.zeroExTransactionHash.length).to.be.eq(66); // tslint:disable-line:custom-no-magic-numbers
+        const threeSecondsMS = ONE_SECOND_MS * 3; // tslint:disable-line:custom-no-magic-numbers
+        const lowerBound = new BigNumber(Date.now() + TEN_MINUTES_MS - threeSecondsMS)
+            .div(ONE_SECOND_MS)
+            .integerValue(BigNumber.ROUND_CEIL);
+        const upperBound = new BigNumber(Date.now() + TEN_MINUTES_MS)
+            .div(ONE_SECOND_MS)
+            .integerValue(BigNumber.ROUND_CEIL);
+        expect(testCase.quote.zeroExTransaction.expirationTimeSeconds).to.be.bignumber.gte(lowerBound);
+        expect(testCase.quote.zeroExTransaction.expirationTimeSeconds).to.be.bignumber.lte(upperBound);
+
+        // NOTE(jalextowle): We pick only the elements that should be tested
+        // against. This avoids altering the original object and running into
+        // an edge-case in `expect` around values defined as `undefined`.
+        expect({
+            price: testCase.quote.price,
+            zeroExTransaction: {
+                signerAddress: testCase.quote.zeroExTransaction.signerAddress,
+                domain: testCase.quote.zeroExTransaction.domain,
+            },
+            orders: testCase.quote.orders,
+            buyAmount: testCase.quote.buyAmount,
+            sellAmount: testCase.quote.sellAmount,
+            sources: testCase.quote.sources,
+        }).to.be.eql({
+            price: testCase.expectedPrice,
+            zeroExTransaction: {
+                signerAddress: takerAddress,
+                domain: { chainId, verifyingContract: contractAddresses.exchange },
+            },
+            orders: testCase.expectedOrders.map(order => stringifyOrderBigNumbers(order)),
+            buyAmount: testCase.expectedBuyAmount,
+            sellAmount: calculateSellAmount(testCase.expectedBuyAmount, testCase.expectedPrice),
+            // NOTE(jalextowle): 0x is the only source that is currently being tested.
+            sources: [
```

Comment:
suggestion: pull this out and `expect(test.quote.sources) to include` so when we add a new source we don't have to update the sources list with a proportion of 0.